define p only when NLS is enabled, otherwise it's not used and build fails:
| cc1: warnings being treated as errors
| main.c: In function 'main':
| main.c:890:8: error: unused variable 'p'
| make[2]: *** [main.o] Error 1
| make[2]: Leaving directory `/OE/tmpdir-shr/work/armv4t-oe-linux-gnueabi/linphone-3.1.0-r3/linphone-3.1.0/gtk-glade'

diff -uNr linphone-3.1.0.orig//gtk-glade/main.c linphone-3.1.0/gtk-glade/main.c
--- linphone-3.1.0.orig//gtk-glade/main.c	2009-03-16 13:12:43.000000000 +0100
+++ linphone-3.1.0/gtk-glade/main.c	2011-05-09 09:53:34.877676870 +0200
@@ -887,7 +887,9 @@
 }
 
 int main(int argc, char *argv[]){
+#ifdef ENABLE_NLS
 	void *p;
+#endif
 	const char *config_file;
 	const char *lang;
 
